src/Client/Decorator/OAuth2/Token/InMemoryAccessTokenCache.php

```diff
@@ -13,16 +13,15 @@
 
 namespace Artemeon\HttpClient\Client\Decorator\OAuth2\Token;
 
+use http\Exception\RuntimeException;
+
 /**
  * Class to store AccessToken in memory
  */
 class InMemoryAccessTokenCache implements AccessTokenCache
 {
-    /** @var AccessToken */
-    private $token;
-
-    /** @var int */
-    private $expireTime;
+    private ?AccessToken $token = null;
```

Comment:
```suggestion
    private AccessToken $token;
```
wenn du Zeile 52 auf `if (!isset($this->token)) {` anpasst, muss die Property nicht mehr _nullable_ sein und du kannst dir die Zuweisung des Default-Werts sparen

Reply:
Das geht leider nicht, weil sonst eine accessed before initialization Fehler auftritt: Fatal error: Uncaught Error: Typed property Artemeon\HttpClient\Client\Decorator\OAuth2\Token\InMemoryAccessTokenCache::$token must not be accessed before initialization in /http-client

Reply:
ok? ich habe evtl. nicht den gesamten Code im Blick gehabt, aber das kann nicht sein? kannst du mir, wenn wir das auch ggf. nicht mehr in diesem PR lösen, trotzdem den Rest der abgeschnittenen Fehlermeldung einfach mal zukommen lassen? soweit ich es sah, gibt es nur genau zwei Zugriffsstellen auf die Property, eine schreibende (unkritisch) und eine lesende (in Zeile 52), die ja entsprechend meines Kommentars nach Umbau auf `isset` auch mit einer nicht-_nullable_-Property funktioniert 🤷‍♂️